### PR TITLE
[FORMS-9142] implement checkbox group component

### PR DESCRIPTION
**Pull request by Aabha1**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { withRuleEngine } from '../shared/withRuleEngine';
+import { PROPS } from '../utils/types';
+import { Checkbox, FormControl, FormControlLabel, FormGroup, FormHelperText, FormLabel } from '@mui/material';
+
+const CheckboxGroupComponent = (props: PROPS) => {
+  const {
+    label, id, required, enum: enums, enabled, errorMessage,
+    description, enumNames, isError, properties
+  } = props;
+  const k = 'afs:layout';
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/CheckboxGroup.tsx:11]: k?

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+  let options = [{ value: enums ? enums[0] : '', label: enumNames ? enumNames[0] : enums ? enums[0] : '' }];
+
+  if (enums) {
+    for (let i = 1; i < enums.length; i++) {
+      let ob = { value: enums[i], label: enumNames ? enumNames[i] : enums[i] };
+      options = [...options, ob];
+    }
+  }
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/CheckboxGroup.tsx:29]: declare options variable like this - const options = enumNames?.length ? enumNames : enums;
https://github.com/adobe/aem-forms-headless-components/blob/main/packages/react-native/src/components/CheckboxGroup.tsx#L13

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    description, enumNames, isError, properties
+  } = props;
+  const k = 'afs:layout';
+  let row = false;
+  if (properties && properties[k]) {
+    row = properties[k].orientation === 'horizontal';
+  }
+
+  const changeHandler = useCallback((event: any) => {
+    props.dispatchChange(event.target.value);
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/CheckboxGroup.tsx:18]: please check this method https://github.com/adobe/aem-forms-headless-components/blob/main/packages/react-native/src/components/CheckboxGroup.tsx#L15

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  return (
+    <FormControl
+      variant="outlined"
+      required={required}
+      disabled={!enabled}
+      fullWidth
+    >
+      <FormLabel error={isError}>
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/CheckboxGroup.tsx:38]: add this condition on FormLabel

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        {label?.visible ? label.value : ''}
+      </FormLabel>
+      <FormGroup row={row} onChange={changeHandler} id={id}>
+        {options.map((option, index) => {
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/CheckboxGroup.tsx:42]: please check this https://github.com/adobe/aem-forms-headless-components/blob/main/packages/react-native/src/components/CheckboxGroup.tsx#L31

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+          return (
+            <FormControlLabel
+              value={option.value}
+              name={option.label}
+              key={index}
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/CheckboxGroup.tsx:47]: can not pass index in key

**Review by amit-mnnit — CHANGES_REQUESTED** [review]
Please incorporate all the comments and also test localization

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx (hunk 1 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/CheckboxGroup.tsx:11]: Removed k

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx (hunk 4 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/CheckboxGroup.tsx:38]: Added the condition

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx (hunk 2 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/CheckboxGroup.tsx:29]: Done as suggested

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx (hunk 3 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/CheckboxGroup.tsx:18]: Checked the method. Improved as required.

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx (hunk 5 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/CheckboxGroup.tsx:42]: Rendered it accordingly.

File: packages/material-ui-based-components/src/components/CheckboxGroup.tsx (hunk 6 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/CheckboxGroup.tsx:47]: Yes, changed it to enums[index]

**Pull request merged**